structuralcodes/codes/ec2_2023/_section_8_1_bending.py

```diff
@@ -0,0 +1,593 @@
+
+    Returns:
+        float: The effectiveness factor kconf_b.
+    """
+    if bcsx < 0:
+        raise ValueError(f'bcsx must be non-negative. Got {bcsx} instead.')
+    if bcsy < 0:
+        raise ValueError(f'b must be non-negative. Got {bcsy} instead.')
+    for value in b_i:
+        if value < 0:
+            raise ValueError(f'b_i must be non-negative. Got {b_i} instead.')
+    if bx < 0:
+        raise ValueError(f'bx must be non-negative. Got {bx} instead.')
+    if by < 0:
+        raise ValueError(f'by must be non-negative. Got {by} instead.')
+
+    sq_sum = 0
+    for item in b_i:
+        sq_sum += item**2
+
+    return (bcsx * bcsy - (1 / 6) * sq_sum) / (bx * by)
+
+
+def kconf_s_multiple(s: float, bcsx: float, bcsy: float) -> float:
+    """Calculate the kconf_s effectiveness factor for square and rectangular
+    members in compression with multiple confinement reinforcement.
+
+    EN1992-1-1:2023 Table (8.1).
+
+    Args:
+        s (float): Spacing of the confinement reinforcement in mm.
+        bcsx (float): Width of the confined section in x-direction in mm.
+        bcsy (float): Width of the confined section in y-direction in mm.
+
+    Returns:
+        float: The effectiveness factor kconf_s.
+    """
+    if s < 0:
+        raise ValueError(f's must be non-negative. Got {s} instead.')
+    if bcsx < 0:
+        raise ValueError(f'bcsx must be non-negative. Got {bcsx} instead.')
+    if bcsy < 0:
+        raise ValueError(f'bcsy must be non-negative. Got {bcsy} instead.')
+
+    return max((1 - s / (2 * bcsx)), 0) * max((1 - s / (2 * bcsy)), 0)
+
+
+def kconf_b_bending(Ac_conf: float, Acc: float, b_i: List[float]) -> float:
+    """Calculate the kconf_b effectiveness factor for compression zones due to
+    bending and axial force.
+
+    EN1992-1-1:2023 Table (8.1).
+
+    Args:
+        Ac_conf (float): Confined area within the centrelines of the
+            confinement reinforcement and the neutral axis in mm2.
+        Acc (float): Compressive area mm2.
+        b_i (List[float]): Distances between bends of straight segments in mm.
+
+    Returns:
+        float: The effectiveness factor kconf_b.
+    """
+    if Ac_conf < 0:
+        raise ValueError(
+            f'Ac_conf must be non-negative. Got {Ac_conf} instead.'
+        )
+    if Acc < 0:
+        raise ValueError(f'Acc must be non-negative. Got {Acc} instead.')
+    for value in b_i:
+        if value < 0:
+            raise ValueError(f'b_i must be non-negative. Got {b_i} instead.')
+
+    sq_sum = 0
+    for item in b_i:
+        sq_sum += item**2
+
+    return (Ac_conf - (1 / 6) * sq_sum) / Acc
+
+
+def kconf_s_bending(s: float, xcs: float, bcsx: float, bcsy: float) -> float:
+    """Calculate the kconf_s effectiveness factor for compression zones due to
+    bending and axial force.
+
+    EN1992-1-1:2023 Table (8.1).
+
+    Args:
+        s (float): Spacing of the confinement reinforcement.
+        xcs (float): Distance to the neutral axis.
```

Comment:
The string is not clear perhaps? But also, I don't understand why it is assumed in this direction